public/MainSerice.js

```diff
@@ -0,0 +1,48 @@
+angular.module('MainService', []).factory('Main', ['$http', function($http) {
+
+  return {
+    //call to get state - on setInterval?
+    getState: function() {
+      return $http.get('/api/vote');
+    },
+
+    //used from view1, sends number of voters and starts the voting (sets vote to view2a)
+    startVoting: function(voterData) {
+      return $http.post('/api/vote', voterData);
+    },
+
+    //sends a Yes or No vote
+    addVoteYes: function(vote) {
+      return $http({
+        method: 'POST',
+        url: '/api/vote/yes',  //vote/yes or 
+        data: vote
+      });
+    },
+
+    //sends a Yes or No vote
+    addVoteNo: function(vote) {
+      return $http({
+        method: 'POST',
+        url: '/api/vote/no',
+        data: vote
+      });
+    },
+
+
+    //Cancel/Reset - not sure if this sends an actual reset vote object, or a message for server to reset the object
+    resetState: function(resetData) {
```

Comment:
If you make the url 'api/vote/reset/', the reset will be handled on the server side.